src/ethereum/cancun/fork.py

```diff
@@ -525,7 +526,14 @@ def apply_body(
         traces=[],
     )
 
-    process_message_call(system_tx_message, system_tx_env)
+    system_tx_output = process_message_call(system_tx_message, system_tx_env)
+
+    destroy_touched_empty_accounts(
+        system_tx_env.state, system_tx_output.touched_accounts
+    )
+
+    if account_exists_and_is_empty(state, BEACON_ROOTS_ADDRESS):
+        destroy_account(state, BEACON_ROOTS_ADDRESS)
```

Comment:
Hm, is it necessary to check again after running `destroy_touched_empty_accounts`?

Reply:
Oh. This is an error. Will update.